sql/src/main/java/io/druid/sql/calcite/aggregation/ApproxCountDistinctSqlAggregator.java

```diff
@@ -82,7 +85,13 @@ public Aggregation toDruidAggregation(
     if (rowSignature.getColumnType(rex.getColumn()) == ValueType.COMPLEX) {
       aggregatorFactory = new HyperUniquesAggregatorFactory(name, rex.getColumn());
     } else {
-      final DimensionSpec dimensionSpec = rex.toDimensionSpec(rowSignature, null);
+      final SqlTypeName sqlTypeName = rexNode.getType().getSqlTypeName();
+      final ValueType outputType = RowSignature.getValueTypeForSqlTypeName(sqlTypeName);
+      if (outputType == null) {
+        throw new ISE("Cannot translate sqlTypeName[%s] to Druid type for field[%s]", sqlTypeName, name);
+      }
+
+      final DimensionSpec dimensionSpec = rex.toDimensionSpec(rowSignature, null, null);
```

Comment:
Better to pass `ValueType.STRING` in here explicitly and check for non-null outputType in toDimensionSpec.